CrmWebApi/Repositories/ProductRepository.cs

```diff
@@ -12,7 +15,8 @@ public class ProductRepository : IProductRepository
 
         public ProductRepository(DatabaseContext databaseContext)
         {
-            _databaseContext = databaseContext;
+            GuardClauses.IsNotNull(databaseContext, nameof(databaseContext));
+            _databaseContext = databaseContext ?? throw new ArgumentNullException(nameof(databaseContext));
```

Comment:
if your're throwing exception in Guard clause, then no need to do that on the next line:
```suggestion
            _databaseContext = databaseContext;
```

Reply:
Yeah, I've seen that later, but haven't commited yet)
Bro, it's Sunday) Stop working :)